Unify internal observability documentation and move it to website (#4192)
open-telemetry/opentelemetry.io: content/en/docs/collector/internal-telemetry.md

### Critical monitoring

#### Data loss

Use rate of `otelcol_processor_dropped_spans > 0` and
`otelcol_processor_dropped_metric_points > 0` to detect data loss, depending on
the requirements set up a minimal time window before alerting, avoiding
notifications for small losses that are not considered outages or within the
desired reliability level.

#### Low on CPU resources

Monitoring CPU resources depends on the CPU metrics available on the deployment.
For example, a Kubernetes deployment may include
`kube_pod_container_resource_limits{resource="cpu", unit="core"}`. Let's call it
`available_cores` below. The idea here is to have an upper bound of the number
of available cores, and the maximum expected ingestion rate considered safe,
let's call it `safe_rate`, per core. This should trigger increase of resources/
instances (or raise an alert as appropriate) whenever
`(actual_rate/available_cores) < safe_rate`.

The `safe_rate` depends on the specific configuration being used.

<!--- TODO: (Carry-over) Provide reference `safe_rate` for a few selected configurations. --->

### Secondary monitoring

#### Queue length

Most exporters offer a

> [queue/retry mechanism](../exporter/exporterhelper/README.md) that is

Author: Fixed. Thanks!

clients. Depending on the deployment and the client’s resilience this may
indicate data loss at the clients.

Sustained rates of `otelcol_exporter_send_failed_spans` and
`otelcol_exporter_send_failed_metric_points` indicate that the Collector is not
able to export data as expected. It doesn't imply data loss per se since there
could be retries but a high rate of failures could indicate issues with the
network or backend receiving the data.

### Data flow

<!--- TODO: Add explanation on using metrics listed in Types? --->

### Logs

Logs can be helpful in identifying issues. Always start by checking the log
output and looking for potential issues. The verbosity level defaults to `INFO`
and can be adjusted.

Set the log level in the config `service::telemetry::logs`

```yaml
service:
  telemetry:
    logs:
      level: 'debug'
```

### Metrics

Prometheus metrics are exposed locally on port `8888` and path `/metrics`. For
containerized environments it may be desirable to expose this port on a public
interface instead of just locally.

Set the address in the config `service::telemetry::metrics`

```yaml
service:
  telemetry:
    metrics:
      address: ':8888'
```

> A Grafana dashboard for these metrics can be found
> [here](https://grafana.com/grafana/dashboards/15983-opentelemetry-collector/).

Reviewer: ```suggestion
To visualize these metrics, you can use the [Grafana dashboard](https://grafana.com/grafana/dashboards/15983-opentelemetry-collector/), for example.
```

extensions:
  zpages:
    endpoint: 0.0.0.0:55679
```

### Local exporters

[Local exporters](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter#general-information)
can be configured to inspect the data being processed by the Collector.

For live troubleshooting purposes consider leveraging the `debug` exporter,
which can be used to confirm that data is being received, processed and exported
by the Collector.

```yaml
receivers:
  zipkin:
exporters:
  debug:
service:
  pipelines:
    traces:
      receivers: [zipkin]
      processors: []
      exporters: [debug]
```

Get a Zipkin payload to test. For example create a file called `trace.json` that
contains:

```json
[
  {
    "traceId": "5982fe77008310cc80f1da5e10147519",
    "parentId": "90394f6bcffb5d13",
    "id": "67fae42571535f60",
    "kind": "SERVER",
    "name": "/m/n/2.6.1",
    "timestamp": 1516781775726000,
    "duration": 26000,
    "localEndpoint": {
      "serviceName": "api"
    },
    "remoteEndpoint": {
      "serviceName": "apip"
    },
    "tags": {
      "data.http_response_code": "201"
    }
  }
]
```

With the Collector running, send this payload to the Collector. For example:

> ```console
> $ curl -X POST localhost:9411/api/v2/spans -H'Content-Type: application/json' -d @trace.json

Reviewer: ```suggestion
```sh
curl -X POST localhost:9411/api/v2/spans -H'Content-Type: application/json' -d @trace.json
```
